iquip/apps/stage.py

```diff
@@ -170,3 +196,143 @@ def __getattr__(self, name: str) -> Callable:
         """
         signal = getattr(self.manager, name)
         return functools.partial(signal.emit, self.key)
+
+
+class StageWidget(QWidget):  # pylint: disable=too-many-instance-attributes
+    """UI for stage control.
+
+    Attributes:
+        connectionButton: Button for toggling rpc connection.
+        positionBox: Spinbox displaying the current position (read-only).
+        absoluteBox: Spinbox for absolute move destination.
+        absoluteButton: Button for absolute move.
+        relativeBox: Spinbox for relative move step size.
+        relativePositiveButton: Button for relative move in positive direction.
+        relativeNegativeButton: Button for relative move in negative direction.
+
+    Signals:
+        moveTo(position_m): Absolute move button is clicked, with the destination
+          position in meters.
+        moveBy(displacement_m): Relative move button is clicked, with the desired
+          displacement in meters.
+        openTarget(): Open button is clicked.
+        closeTarget(): Close button is clicked.
+    
+    All the displayed values are in mm unit.
+    However, the values for interface (methods and signals) are in m.
+    """
+
+    moveTo = pyqtSignal(float)
+    moveBy = pyqtSignal(float)
+    openTarget = pyqtSignal()
+    closeTarget = pyqtSignal()
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        # widgets
+        self.connectionButton = QPushButton("Open", self)
+        self.connectionButton.setCheckable(True)
+        self.positionBox = QDoubleSpinBox(self)
+        self.positionBox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+        self.positionBox.setReadOnly(True)
+        self.positionBox.setDecimals(3)
+        self.positionBox.setSuffix("mm")
+        self.positionBox.setAlignment(Qt.AlignHCenter)
+        self.absoluteBox = QDoubleSpinBox(self)
+        self.absoluteBox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+        self.absoluteBox.setDecimals(3)
+        self.absoluteBox.setSingleStep(0.001)
```

Comment:
Is there any reason to set its single step even if there is no button?

Reply:
We can adjust the value by scrolling mouse wheel or pressing keyboard arrow keys.

Reply:
Oh, I forgot we can adjust it without clicking the button directly.
Then, please never mind😃